src/librustc_resolve/lib.rs

```diff
@@ -3273,6 +3274,38 @@ impl<'a> Resolver<'a> {
                     let traits = self.get_traits_containing_item(item_name, ns);
                     self.trait_map.insert(id, traits);
                 }
+
+                // check if we wrote `str::from_utf8` instead of `std::str::from_utf8`
+                let item_span = path.iter().last().map(|segment| segment.ident.span)
+                    .unwrap_or(span);
```

Comment:
Nit: could you move this closer to the use point (which will also avoid evaluating the span unless necessary).